src/fu-engine.c

```diff
@@ -5254,9 +5254,9 @@ fu_engine_load (FuEngine *self, FuEngineLoadFlags flags, GError **error)
 /* TODO: Read registry key [HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography] "MachineGuid" */
 #ifndef _WIN32
 	/* cache machine ID so we can use it from a sandboxed app */
-	self->host_machine_id = fwupd_build_machine_id ("fwupd", error);
+	self->host_machine_id = fwupd_build_machine_id ("fwupd", NULL);
```

Comment:
If it does end up being shown as a warning a better way to do it is to declare a local error like this:

```
g_autoptr(GError) error_machine_id = NULL;
self->host_machine_id =fwupd_build_machine_id ("fwupd", &error_local);
if (self->host_machine_id == NULL)
      g_warning ("%s", error_local->message);
```

Reply:
Is there anything better we can do? e.g. have a "fake" machine ID for chrome-os?

Reply:
Oh I like that; so maybe add a new path only present on chrome systems that it can check for and set that fake string.

Reply:
I proposed this yesterday to be handled outside fwupd code base and got this: crrev.com/c/2078258/12#message-e151fbe95c08c57c5228904c7d1ddeac151ab7dd

Reply:
For some reason that's not showing an actual message.  I don't think it needs to be outside the fwupd code base.  I'm saying in the fwupd codebase we can add an extra thing it falls back on if `/etc/machine-id` is missing at all those above locations that can indicate it's a chrome system and to use a dummy value instead.

Another alternative to do is if nothing is found then instead use one of the existing helpers that parsed `/etc/os-release` and fill in machine-id with the `ID` key from there.

Reply:
Maybe the shortening magic... here again: https://chromium-review.googlesource.com/c/chromiumos/overlays/chromiumos-overlay/+/2078258/12#message-e151fbe95c08c57c5228904c7d1ddeac151ab7dd

This is a sample os-release in my device:
BUILD_ID=12735.0.2019_12_06_1141
NAME=Chrome OS
ID_LIKE=chromiumos
GOOGLE_CRASH_ID=ChromeOS
VERSION_ID=80
BUG_REPORT_URL=https://crbug.com/new
VERSION=80
HOME_URL=https://www.chromium.org/chromium-os
ID=chromeos

ID would just be chromeos... It will probably be meaningless to use this.